e2e_node: fix DevicePlugin feature flags

bart0sh opened the pull request:
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Feature:DevicePluginProbe and NodeFeature:DevicePluginProbe are not used by any of the test-infra jobs.
    
This commit renames NodeFeature:DevicePluginProbe to NodeFeature:DevicePlugin and removes Feature:DevicePlugin and Feature:DeviceManager to avoid having both Feature and NodeFeature tags for the same feature.
    
NOTE: Test-infra SIG-Node jobs should focus on the NodeFeature:DevicePlugin to run generic Device Plugins tests.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

bart0sh commented:
/cc @ffromani @swatisehgal @SergeyKanzhelev

bart0sh commented:
@swatisehgal It could be that I misunderstood which DevicePlugin e2e tests should be running by which jobs. Can you point me out to the jobs(s) running generic [device plugins](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/device_plugin_test.go) and [device manager](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/device_manager_test.go) tests?

It looks like `Feature:GPUDevicePlugin` is for running [GPU-specific tests](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scheduling/nvidia-gpus.go#L228) and we should use something else (`NodeFeature:DevicePlugin` ?) to run generic device plugin tests.

bart0sh commented:
@kannon92 @SergeyKanzhelev PTAL

bart0sh commented:
@SergeyKanzhelev To your question about real GPU jobs. There is at least one: https://testgrid.k8s.io/sig-release-master-blocking#gce-device-plugin-gpu-master and it seems to run one GPU test: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scheduling/nvidia-gpus.go#L327.

Additionally it runs windows test case for DevicePlugin, but this should be fixed by this PR.

kannon92 commented:
/test

kannon92 commented:
code looks good to me. Is there any jobs you can trigger to verify that this behaves as you expect?

bart0sh commented:
@kannon92 I don't know any presubmit jobs that use either GPUDevicePlugin or DevicePluginProbe features and we can't trigger ci- jobs here, right?

bart0sh commented:
@kannon92 I think I found it. DevicePlugin tests are serial, so this one should include them:
/test pull-e2e-serial-ec2

bart0sh commented:
/test pull-kubernetes-node-e2e-containerd-serial-ec2

bart0sh commented:
This one should run all Device Plugins and Device Manager tests on CRI-O:
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv2

So, I was wrong - we run all generic Device Plugins and Device Manager tests for both runtimes as part of Serial jobs.

kannon92 commented:
/lgtm
/assign @SergeyKanzhelev @klueska

bart0sh commented:
/retest
/test pull-kubernetes-node-kubelet-serial-crio-cgroupv2
/test pull-kubernetes-node-e2e-containerd-serial-ec2

bart0sh commented:
/triage accepted
/priority important-longterm

bart0sh commented:
removed feature.DevicePlugin and feature.DeviceManager definitions as they're not used anywhere in the test. nodefeature.DevicePlugin and nodefeature.DeviceManager should be enough in my opinion.

@kannon92 @SergeyKanzhelev PTAL, thanks!

bart0sh commented:
/retest

bart0sh committed:
e2e_node: fix DevicePlugin feature flags

Feature:DevicePluginProbe and NodeFeature:DevicePluginProbe
are not used by any of the test-infra jobs.

This commit renames NodeFeature:DevicePluginProbe to NodeFeature:DevicePlugin
and removes Feature:DevicePlugin and Feature:DeviceManager to avoid
having both Feature and NodeFeature tags for the same feature.

NOTE: Test-infra SIG-Node jobs should focus on the
NodeFeature:DevicePlugin to run generic Device Plugins tests.

bart0sh commented:
/retest

bart0sh commented:
/remove-kind bug

bart0sh commented:
/remove-sig windows

kannon92 commented:
/lgtm

SergeyKanzhelev reviewed (approved):
/approve
/lgtm

SergeyKanzhelev commented on test/e2e/nodefeature/nodefeature.go:
```diff
@@ -38,7 +38,7 @@ var (
 	DeviceManager = framework.WithNodeFeature(framework.ValidNodeFeatures.Add("DeviceManager"))
 
 	// TODO: document the feature (owning SIG, when to use this feature for a test)
```
please fill up the TODO while you are on it

bart0sh replied:
Will do in a separate PR. There are the same TODO comments for all the features in this file.

bart0sh commented:
@kannon92 @SergeyKanzhelev @klueska rebased, PTAL

haircommander commented:
/lgtm

bart0sh commented:
@SergeyKanzhelev @dims @mrunalp can you review and approve?

dims commented:
/approve
/lgtm

Nomini37 commented:
> #### What type of PR is this?
> /kind cleanup
> 
> #### What this PR does / why we need it:
> Feature:DevicePluginProbe and NodeFeature:DevicePluginProbe are not used by any of the test-infra jobs.
> 
> This commit renames NodeFeature:DevicePluginProbe to NodeFeature:DevicePlugin and removes Feature:DevicePlugin and Feature:DeviceManager to avoid having both Feature and NodeFeature tags for the same feature.
> 
> NOTE: Test-infra SIG-Node jobs should focus on the NodeFeature:DevicePlugin to run generic Device Plugins tests.
> 
> #### Does this PR introduce a user-facing change?
> ```
> NONE
> ```

Ddd

Nomini37 commented:
Ok
